File: frontend/src/types/review.ts
export interface Review {
  id: number;
  customer_name: string;
  content: string;
  rating: number;
  service_category: string | null;
  review_date: string | null;
  featured: boolean;
  active: boolean;
  position: number;
  created_at: string;
}

File: backend/app/models/review.rb
class Review < ApplicationRecord
  before_validation :sanitize_inputs

  validates :customer_name, presence: true
  validates :content, presence: true
  validates :rating, presence: true, inclusion: { in: 1..5 }

  scope :active, -> { where(active: true) }
  scope :featured, -> { where(featured: true) }
  scope :ordered, -> { order(position: :asc, created_at: :desc) }

  private

  def sanitize_inputs
    self.customer_name = InputSanitizer.sanitize_text(customer_name) if customer_name.present?
    self.content = InputSanitizer.sanitize_text(content) if content.present?
    self.service_category = InputSanitizer.sanitize_text(service_category) if service_category.present?
  end
end
